controller/src/beerocks/master/db/db.cpp

```diff
@@ -2117,6 +2117,324 @@ bool db::get_hostap_is_acs_enabled(std::string mac)
+}
+
+bool db::set_channel_scan_dwell_time_msec(const std::string &mac, int dwell_time_msec,
+                                          bool single_scan)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+    LOG(DEBUG) << __FUNCTION__ << "," << (single_scan ? "single" : "continuous")
+               << ", dwell time msec = " << dwell_time_msec;
+
+    if (dwell_time_msec <= 0) {
+        LOG(ERROR) << __FUNCTION__ << "Only >0 dwell time is supported!";
+        return false;
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+    scan_config->dwell_time_msec = dwell_time_msec;
+
+    return true;
+}
+
+int db::get_channel_scan_dwell_time_msec(const std::string &mac, const bool single_scan)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return -1;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return -1;
+    }
+
+    auto scan_config =
+        (single_scan) ? n->hostap->single_scan_config : n->hostap->continuous_scan_config;
+    return scan_config->dwell_time_msec;
+}
+
+bool db::set_channel_scan_pool(const std::string &mac, const std::set<uint8_t> &channel_pool,
+                               const bool single_scan)
+{
+    auto n = get_node(mac);
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+
+    //Validate new dcs channel pool with hostap_supported_channels list
+    //This is a workaround
+    static uint8_t supported_channels_array[] = {
+        1,  2,   3,   4,   5,   6,   7,   8,   9,   10,  11,  36,  40,  44,  48,  52,  56,  60,
+        64, 100, 104, 108, 112, 116, 120, 124, 128, 132, 136, 140, 144, 149, 153, 157, 161, 165};
+    static size_t supported_channels_array_size =
+        sizeof(supported_channels_array) / sizeof(uint8_t);
+    std::set<uint8_t> supported_channels(supported_channels_array,
+                                         supported_channels_array + supported_channels_array_size);
+    for (auto channel : channel_pool) {
+        if (supported_channels.find(channel) == supported_channels.end()) {
+            //even if one channel is not supported reject all pool
+            LOG(ERROR) << __FUNCTION__ << "channel =" << int(channel)
+                       << " is invalid, setting channel pool failed !";
+            return false;
+        }
+    }
```

Comment:
What's the purpose of this test?
It seems like you build a list of ALL available channels, and just check that the given channel is in this list.
Do we expect here just a valid channel or we need to make sure that the Agent actually supports this channel?

Reply:
When setting the channel pool, we need to make sure that all the channels in the new pool are supported, we tried using existing lists in the db to compare to, but we couldn't find any that did so we defaulted to a static list of the supported channels

Reply:
The supported channels should come from BWL - and not declared as const. #665

Reply:
opened separate task for this 
see https://github.com/prplfoundation/prplMesh/pull/643#discussion_r367404728